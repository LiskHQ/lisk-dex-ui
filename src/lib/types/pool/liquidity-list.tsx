export interface LiuqidityListInterface {
    token1Id: string;
    token2Id: string;
    feeTier: string;
    pooledtoken1: string;
    pooledtoken2: string;
    poolTokens: string;
    poolShare: string;
    accumulatedFeesToken1: string;
    accumulatedFeesToken2: string;
    totalFees: string;
  }