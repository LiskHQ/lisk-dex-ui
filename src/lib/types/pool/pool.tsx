export interface LiquidityListInterface {
  token1Id: string;
  token2Id: string;
  feeTier: string;
  pooledtoken1: string;
  pooledtoken2: string;
  poolTokens: string;
  poolShare: string;
  accumulatedFeesToken1: string;
  accumulatedFeesToken2: string;
  totalFees: string;
}

export interface IpriceRange {
  minPrice: string,
  maxPrice: string,
  currentPrice: string  
}

export  interface IdepositAmounts {
  token1: string,
  token2: string,
  amount1: string,
  amount2: string,
  feeTier: string,
}
