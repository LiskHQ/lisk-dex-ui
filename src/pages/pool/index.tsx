import { NextPage } from 'next';
import { PoolContainer } from 'containers';
import { withLayout } from 'components';

const PoolPage: NextPage = () => {
  return (
    <PoolContainer />
  );
};

export default withLayout(PoolPage);