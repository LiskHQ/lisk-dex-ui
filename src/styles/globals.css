:root {
  font-size: 16px;
}

html,
body,
#__next {
  padding: 0 !important;
  margin: 0;
  height: 100%;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

a,
.btn,
button,
span,
p,
i,
input,
select,
textarea,
li,
img,
svg path,
*::after,
*::before,
.transition-3,
h1,
h2,
h3,
h4,
h5,
h6 {
  transition: all 0.3s ease-out 0s;
}

span,
p,
a,
li.MuiMenuItem-root,
label,
label.MuiFormLabel-root,
span.MuiTypography-root,
p.MuiTypography-root,
button.MuiButton-root,
button.MuiButtonBase-root,
.MuiInput-root,
.MuiTableCell-root {
  font-family: 'Roboto', sans-serif;
}

h1,
h2,
h3,
h4,
h5,
h6,
h1.MuiTypography-root,
h2.MuiTypography-root,
h3.MuiTypography-root,
h4.MuiTypography-root,
h5.MuiTypography-root,
h6.MuiTypography-root {
  font-family: 'Roboto', sans-serif;
}