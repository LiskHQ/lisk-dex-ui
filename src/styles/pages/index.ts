import { GovernanceStyle } from "./governance/index.style";

export {
  GovernanceStyle
}