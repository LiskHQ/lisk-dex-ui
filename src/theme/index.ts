export { default as theme } from "./theme";
export { default as variables } from "./variables";
