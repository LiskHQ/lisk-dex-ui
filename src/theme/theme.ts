import {
  createTheme,
  responsiveFontSizes,
  ThemeOptions,
} from "@mui/material/styles";

import variables from "./variables";

const options: ThemeOptions = {
  palette: {
    primary: variables.primary,
    secondary: variables.secondary,
  },
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 400,
    },
    h3: {
      fontSize: "1.75rem",
      fontWeight: 400,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 400,
    },
    h5: {
      fontSize: "1rem",
      fontWeight: 400,
    },
    h6: {
      fontSize: "0.75rem",
      fontWeight: 400,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
      xxl: 1950,
      xxxl: 2300,
    },
  },
};

const theme = responsiveFontSizes(createTheme(options));
export default theme;
