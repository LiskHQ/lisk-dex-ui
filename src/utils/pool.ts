export const getPoolToken0 = (pool: string) => {
  return pool.split('-')[0];
};

export const getPoolToken1 = (pool: string) => {
  return pool.split('-')[1];
};