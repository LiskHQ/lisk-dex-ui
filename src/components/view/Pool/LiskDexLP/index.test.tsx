import React from "react";
import { ThemeProvider } from "@mui/material";
import { fireEvent, render } from "@testing-library/react";
import { LiskDexLP, ILiskDexLProps } from "./index";
import { lightTheme } from "styles/theme";
import { mockPools } from "__mock__/pool.mock";

function renderComponent(props: ILiskDexLProps) {
  return render(
    <ThemeProvider theme={lightTheme} >
      <LiskDexLP {...props} />
    </ThemeProvider>
  );
}

describe("LiskDexLP component", () => {
  const mockProps: ILiskDexLProps = {
    pools: mockPools,
    gettingPools: false,
    gotPools: false,
    onPreview: jest.fn(),
    onPreviewRemove: jest.fn(),
  }

  it("checks if the component matches the b  snapshot", () => {
    const { container } = renderComponent(mockProps);
    expect(container).toMatchSnapshot();
  });

  it("click on preview", () => {
    const { getByTestId } = renderComponent(mockProps);
    setTimeout(() => {
      fireEvent.click(getByTestId("pool-component-test-1001"));
      fireEvent.click(getByTestId("increase-liquidity-test"));
      expect(getByTestId("increase-liquidity-modal-test")).toBeInTheDocument();
    }, 1000);
  });
});
