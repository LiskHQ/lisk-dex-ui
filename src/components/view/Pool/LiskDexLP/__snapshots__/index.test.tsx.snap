// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiskDexLP component checks if the component matches the b  snapshot 1`] = `
.emotion-0 {
  position: relative;
  border-radius: 0.5rem;
  box-shadow: 0px 1px 7px rgba(0, 0, 0, 0.1);
  background: #FFFFFF;
  padding: 1.5rem;
}

.emotion-0 .lisk-dex-lp-main {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-top: 2.5rem;
  margin-bottom: 1rem;
}

.emotion-0 .lisk-dex-lp-main .loader-component svg path {
  stroke: #6953F4;
}

.emotion-2 {
  margin: 0;
  font-size: 1.25rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00735em;
}

.emotion-4 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.375rem;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="lisk-dex-lp-title MuiBox-root emotion-1"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 emotion-2"
      >
        Your LiskDEX LP 
      </h4>
    </div>
    <div
      class="lisk-dex-lp-main MuiBox-root emotion-1"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 emotion-4"
      >
        You do not have any liquidity positions.
      </p>
    </div>
  </div>
</div>
`;
